### DISCUSSION TITLE
EXIV2 retrieve value from a single specific IPTC tag

### DISCUSSION THREAD

**Discussion by airdata3** [Exiv2/exiv2#2658] [Q&A] [answered]
I am trying to retrieve a single value from an image's IPTC tag.  I have tried multiple syntax but not getting what I need.  Hopefully someone here can help.

The command line entry: 
exiv2 -piv d:\aviation\photos\2023\23L0010.jpg

This retrieves 12 lines of tags.  I need the value contained in one tag only - Iptc.Application2.Caption
How do I retrieve this using EXIV2 command line?

Any help is very much appreciated.
Thanks

Mark

**Comment by postscript-dev** [comment, marked as answer]
@airdata3 
I don't know which version of Exiv2 you are using but the latest [0.28.0 manpage](https://exiv2.org/manpage.html) only has a couple of [very minor changes](https://github.com/Exiv2/exiv2/commits/0.28.x/exiv2.md) from several 0.27.x releases. The new manpage contains more information and examples.

> The command line entry:
> exiv2 -piv d:\aviation\photos\2023\23L0010.jpg

> This retrieves 12 lines of tags. I need the value contained in one tag only - Iptc.Application2.Caption
> How do I retrieve this using EXIV2 command line?

In the exiv2 app, you can filter specific tags with [--key key](https://exiv2.org/manpage.html#key_key) or groups of tags using [--grep str](https://exiv2.org/manpage.html#grep_str). If you need to, you can control the output more precisely by using [--Print flgs](https://exiv2.org/manpage.html#Print_flgs) instead of [--print mod](https://exiv2.org/manpage.html#print_mod).

It is worth noting that when using Windows' `cmd.exe` or Powershell, [globbing](https://www.definitions.net/definition/globbing) is not supported (see #2633 for a work around). The manpage has not been updated yet to reflect this issue.

**Reply by airdata3** [reply]
Thanks for the reply.  I got it going and got a shell command going from within an Access database to get the required tag information.

**Comment by clanmills** [comment]
As always, @postscript-dev has provided a perfect answer.  Here's my version of the same message:

1.  Get the IPTC metadata:
```
531 rmills@rmillsm1:~/temp $ exiv2 -pi https://clanmills.com/Stonehenge.jpg
Iptc.Envelope.ModelVersion                   Short       1  4
Iptc.Envelope.CharacterSet                   String      3  G
Iptc.Application2.RecordVersion              Short       1  4
Iptc.Application2.Caption                    String     12  Classic View
```
2. Select one tag of interest:
```
532 rmills@rmillsm1:~/temp $ exiv2 -K   Iptc.Application2.Caption  https://clanmills.com/Stonehenge.jpg
Iptc.Application2.Caption                    String     12  Classic View
```
3.  Only print the value of the tag:
```
533 rmills@rmillsm1:~/temp $ exiv2 -K   Iptc.Application2.Caption  -Pv https://clanmills.com/Stonehenge.jpg
Classic View
534 rmills@rmillsm1:~/temp $ 
```
Well done @airdata3 to execute a shell command from access.

Case Closed?
